include/plugins/slotproxy.h

```diff
@@ -75,18 +76,25 @@ class SlotProxy
      */
     constexpr auto generate(auto&&... args)
     {
-        if (plugin_.has_value())
+        if (! plugins_.empty())
         {
-            return plugin_.value().generate(std::forward<decltype(args)>(args)...);
+            return plugins_.front().generate(std::forward<decltype(args)>(args)...);
         }
         return std::invoke(default_process, std::forward<decltype(args)>(args)...);
     }
 
     constexpr auto modify(auto& original_value, auto&&... args)
     {
-        if (plugin_.has_value())
+        if (! plugins_.empty())
         {
-            return plugin_.value().modify(original_value, std::forward<decltype(args)>(args)...);
+            auto modified_value = original_value;
```

Comment:
```suggestion
            auto& modified_value = original_value;
```

Reply:
Well, then it would actually make sense to not return it at all, but input a reference and directly apply the changes on it, what do you think ?

Reply:
Sort-of?

One the one hand, my suggestion is a simple change, that already works without going the full mile, since it avoids the extra copies while constructing the `modified` value and iterating over it copying the return to it each time (though the compiler _might_ optimize some of that).

On the other hand, it _is_ actually building up a new 'return' value (initially empty) in the `modify` call as well, which we might improve by moving/swapping it into the 'original' in the end -- that does require some work though, and may require re-building existing plugins, if we thread this change all the way down.

Reply:
I made a commit that seems to work, and actually avoids a few copies, which is a good thing because the objects in question can be very big (list of gcode paths of a whole layer).

However I have a suspicion of possible behavior change in `converters.cpp`. Tell me if you catch something obvious, otherwise I'll share more details :smiley: